lib/chef/knife/ssh.rb

```diff
@@ -134,6 +134,12 @@ class Ssh < Knife
         boolean: true,
         default: false
 
+      option :ssh_pty,
+        long: "--ssh-pty",
+        description: "PTY flag. Provide this option if PTY is configured on node.",
+        boolean: true,
+        default: false
```

Comment:
this will probably break users that use sudo on linux by turning this off by default.  the default requesting of a pty is probably right for windows.  the problem is that it depends on the target host.  if we used train here we could detect the remote O/S.  most likely the sane default is to set this to true and then require windows users to set it to false when they need to.  maybe wrap the use case with a timeout and in the event of a timeout, warn the user that they should use --no-ssh-pty and move on without it.

Reply:
@lamont-granquist I would suggest here to catching the exception rather than providing an option because if it returns multiple nodes and we are appending `pty: true` or `pty: false` for all nodes which definitely breaking changes.

Reply:
Thank you @lamont-granquist & @vsingh-msys removing `--ssh-pty` entirely.